Tests/SwiftIDEUtilsTest/ClassificationTests.swift

```diff
@@ -121,4 +121,379 @@ public class ClassificationTests: XCTestCase {
+        ClassificationSpec(source: "#else", kind: .poundDirective),
+        ClassificationSpec(source: "#endif", kind: .poundDirective)
+      ]
+    )
+  }
+  
+  public func testIfconfig2() {
+    assertClassification(
+      """
+      #if !CONF
+      #else
+      """,
+      expected: [
+        ClassificationSpec(source: "#if", kind: .poundDirective),
+        ClassificationSpec(source: "!", kind: .operatorIdentifier),
+        ClassificationSpec(source: "CONF", kind: .buildConfigId),
+        ClassificationSpec(source: "#else", kind: .poundDirective),
+      ]
+    )
+  }
+  
+  public func testBuiltinDirectives() {
+    // https://github.com/apple/swift-syntax/issues/1044
+    XCTExpectFailure("the directive keywords are classied as # and identifier")
+    assertClassification(
+      """
+      #error("Error")
+      #warning("Warning")
+      #sourceLocation(file: "x", line: 1)
+      """,
+      expected: [
+        ClassificationSpec(source: "#error", kind: .poundDirective),
+        ClassificationSpec(source: "\"Error\"", kind: .stringLiteral),
+        ClassificationSpec(source: "#warning", kind: .poundDirective),
+        ClassificationSpec(source: "\"Warning\"", kind: .stringLiteral),
+        ClassificationSpec(source: "#sourceLocation", kind: .poundDirective),
+        ClassificationSpec(source: "file", kind: .keyword),
+        ClassificationSpec(source: "\"x\"", kind: .stringLiteral),
+        ClassificationSpec(source: "line", kind: .keyword),
+        ClassificationSpec(source: "1", kind: .integerLiteral),
+      ]
+    )
+  }
+  
+  public func testColorLiteral() {
+    // https://github.com/apple/swift-syntax/issues/1044
+    XCTExpectFailure("the directive keywords are classied as # and identifier")
+    let colorLiteral = "#colorLiteral(red: 0, green: 0, blue: 0, alpha: 1)"
+    assertClassification(
+      colorLiteral
+      , expected: [
+        ClassificationSpec(source: colorLiteral, kind: .objectLiteral)
+      ]
+    )
+  }
+  
+  public func testImageLiteral() {
+    // https://github.com/apple/swift-syntax/issues/1044
+    XCTExpectFailure("the directive keywords are classied as a # and an identifier")
+    let imageLiteral = "#imageLiteral(resourceName: \"cloud.png\")"
+    assertClassification(imageLiteral, expected: [ClassificationSpec(source: imageLiteral, kind: .objectLiteral)])
+  }
+  
+  public func testFileLiteral() {
+    // https://github.com/apple/swift-syntax/issues/1044
+    XCTExpectFailure("the directive keywords are classied as # and identifier")
+    let fileLiteral = "#fileLiteral(resourceName: \"cloud.png\")"
+    assertClassification(fileLiteral, expected: [ClassificationSpec(source: fileLiteral, kind: .objectLiteral)])
+  }
+  
+  public func testAttribute() {
+    XCTExpectFailure(
+      """
+      '@available' is classified as @ and a typeIdentifier
+      'iOS' is classified as a typeIdentifier
+      '8.0' and '10.10' are classified as integerLiteral
+      """
+    )
```

Comment:
Hmm, this is expected by the way the syntax tree gets represented because by now we always model the name of an attribute as a type. Also the lines between builtin attributes (like `@availablable`) with completely custom syntax, user-defined attributes that are types (like result builders and property wrappers) and user-defined types that aren’t types (attached macros) are blurring. So, what I would expect is `available` to be classified as an `attribute`. 

You should be able to achieve that by changing `\AttributeSyntax.attributeName` to return `(.attribute, false)` in `SyntaxClassification.classify(_ keyPath:)`

Same for `testAttribute2` where `objc` and `IBOutlet` should be classified as `attribute`.

Reply:
> You should be able to achieve that by changing \AttributeSyntax.attributeName to return (.attribute, false)

Do you mean we changing this to `(.attribute, true)`? Since `\AttributeSyntax.attributeName` already returns `(.attribute, false)` now. 

If so, we may need to make some changes on `handleLayout` (as my new commits) since `\AttributeSyntax.attributeName` is not a `TokenSyntax` and the second return value of `SyntaxClassification.classify(_ keyPath:)` only has effects in `handleToken`.

But I can't find a proper workaround for classifying the trivia pieces in a layout node for now though.

Reply:
> Do you mean we changing this to `(.attribute, true)`? Since `\AttributeSyntax.attributeName` already returns `(.attribute, false)` now.
>
> If so, we may need to make some changes on `handleLayout` (as my new commits) since `\AttributeSyntax.attributeName` is not a `TokenSyntax` and the second return value of `SyntaxClassification.classify(_ keyPath:)` only has effects in `handleToken`.

Yes, that’s what I meant. But I didn’t realize it only had an effect on tokens 😢 

> On a second thought, what do you think about refactor `ClassificationVisitor` as a subclass of `SyntaxAnyVisitor` like `ParseDiagnosticsGenerator`. That should make this case handled easier. And the `8.0` and `10.10` can be specified as `floatingLiteral` rather that some `integerLiteral`s in a version tuple.

I think that would be a good idea and should make the classifier easier to maintain. If the `SyntaxAnyVisitor`-based implementation is significantly slower than the current one, we should look into why that is and see if we can make `SyntaxAnyVisitor` faster.

In any case, I would prefer to do this in a follow-up PR. We can just keep the test case disabled for now. One note though: `XCTExpectFailure` is not supported on Linux IIRC, so you would need to do `try XCTSkipIf(true, <message>)`